target/arc/arc-semi.c

```diff
@@ -117,26 +117,26 @@ struct arc_stat
 	uint32_t my___unused5;
 };
 
-static struct arc_stat conv_stat(struct stat *st);
+static struct arc_stat *conv_stat(struct stat *st);
 
 /* Converts linux's stat to newlib's stat struct */
-static struct arc_stat conv_stat(struct stat *st)
+static struct arc_stat *conv_stat(struct stat *st)
 {
-    struct arc_stat arc_st;
-
-    arc_st.my_dev = st->st_dev;
-    arc_st.my_ino = st->st_ino;
-    arc_st.my_mode = st->st_mode;
-    arc_st.my_nlink = st->st_nlink;
-    arc_st.my_uid = st->st_uid;
-    arc_st.my_gid = st->st_gid;
-    arc_st.my_rdev = st->st_rdev;
-    arc_st.my_size = st->st_size;
-    arc_st.my_blocks = st->st_blocks;
-    arc_st.my_blksize = st->st_blksize;
-    arc_st.my_atime = st->st_atime;
-    arc_st.my_mtime = st->st_mtime;
-    arc_st.my_ctime = st->st_ctime;
+    struct arc_stat *arc_st = malloc (sizeof (struct arc_stat));
```

Comment:
don't we need to check for malloc's return value (NULL)?

Reply:
That's an artifact of the force-push to master just done. I've rebased and updated this branch, so the changes should be easier to see now.

Reply:
That's why I already approved this PR. The comment was more addressed to the original author (@luismgsilva ) which he should fix separately.

Thank you for your contribution @keith-packard !